server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/executeBash.ts

```diff
@@ -146,6 +154,20 @@ export class ExecuteBash {
             throw new Error('No command found.')
         }
 
+        // Extract individual base commands for telemetry purposes
+        const baseCommands = parseBaseCommands(args)
+        baseCommands.forEach(command => {
+            const metricPayload = {
+                name: ChatTelemetryEventName.BashCommand,
+                data: {
+                    credentialStartUrl: this.credentialsProvider.getConnectionMetadata()?.sso?.startUrl,
+                    result: 'Succeeded',
```

Comment:
Shouldn't we emit this in the invoke method? If validation fails, telemetry will still be emitted for commands that never executed.